providers/flagsmith/src/main/java/dev.openfeature.contrib.providers.flagsmith/FlagsmithProvider.java

```diff
@@ -0,0 +1,263 @@
+package dev.openfeature.contrib.providers.flagsmith;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.BooleanNode;
+import com.fasterxml.jackson.databind.node.DoubleNode;
+import com.fasterxml.jackson.databind.node.IntNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.databind.node.TextNode;
+import com.flagsmith.FlagsmithClient;
+import com.flagsmith.exceptions.FlagsmithClientError;
+import com.flagsmith.models.Flags;
+import dev.openfeature.contrib.providers.flagsmith.exceptions.FlagsmithJsonException;
+import dev.openfeature.sdk.ErrorCode;
+import dev.openfeature.sdk.EvaluationContext;
+import dev.openfeature.sdk.FeatureProvider;
+import dev.openfeature.sdk.Hook;
+import dev.openfeature.sdk.Metadata;
+import dev.openfeature.sdk.MutableStructure;
+import dev.openfeature.sdk.ProviderEvaluation;
+import dev.openfeature.sdk.Reason;
+import dev.openfeature.sdk.Structure;
+import dev.openfeature.sdk.Value;
+import dev.openfeature.sdk.exceptions.OpenFeatureError;
+import dev.openfeature.sdk.exceptions.TypeMismatchError;
+import java.time.Instant;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * FlagsmithProvider is the JAVA provider implementation for the feature flag solution Flagsmith.
+ */
+@Slf4j
+class FlagsmithProvider implements FeatureProvider {
+
+    private static final String NAME = "Flagsmith Provider";
+    private static FlagsmithClient flagsmith;
+    private FlagsmithProviderOptions options;
+
+    public FlagsmithProvider(FlagsmithProviderOptions options) {
+        this.options = options;
+        flagsmith = FlagsmithClientConfigurer.initializeProvider(options);
+    }
+
+    @Override
+    public Metadata getMetadata() {
+        return () -> NAME;
+    }
+
+    @Override
+    public List<Hook> getProviderHooks() {
+        return FeatureProvider.super.getProviderHooks();
+    }
```

Comment:
Is there a reason to override the default impl just to call it?

Reply:
Not really, just noticed it had been done in the go-feature-flag provider, I've removed it ow, since it's not needed